edx_arch_experiments/kafka_consumer/docs/decisions/0001_kafka_based_event_bus.rst

```diff
@@ -0,0 +1,154 @@
+
+Decision
+--------
+
+An initial implementation of an event bus for the Open edX platform will be implemented using `Kafka`_. This implementation will be used by edx.org, and available to the Open edX community.
+
+However, given the intention of creating an abstraction layer around the event bus technology, this decision does not preclude the introduction of alternative event bus implementations based on other technologies in the future.
+
+.. _Kafka: https://kafka.apache.org/
+
+Why Kafka?
+~~~~~~~~~~
+
+Kafka is a distributed streaming platform. Kafka's implementation maps nicely to the pub/sub pattern. However, some native features of a message broker are not built-in.
+
+Kafka has been around for a long time. See `Thoughtworks's technology radar introduced Kafka`_ as "Assess" in 2015, and "Trial" in 2016. It never moved up to "Adopt", and also never moved down to "Hold". Read `Thoughtwork's Kafka decoder page`_ to learn more about its benefits and trade-offs, and how it is used.
+
+More recently, the `Thoughtworks's technology radar introduced Apache Pulsar`_ as "assess" in 2020, and the `technology radar introduced Kafka API without Kafka`_ in 2021. This both demonstrates the de facto standard of the Kafka API, but also Thoughtwork's hope to find a less complex alternative.
+
+We believe Apache Kafka is still the right option due to its maturity, documentation, support and community.
+
+.. _Thoughtworks's technology radar introduced Kafka: https://www.thoughtworks.com/radar/tools/apache-kafka
+.. _Thoughtwork's Kafka decoder page: https://www.thoughtworks.com/decoder/kafka
+
+.. _Thoughtworks's technology radar introduced Apache Pulsar: https://www.thoughtworks.com/radar/platforms/apache-pulsar
+.. _technology radar introduced Kafka API without Kafka: https://www.thoughtworks.com/radar/platforms/kafka-api-without-kafka
+
+Kafka Highlights
+~~~~~~~~~~~~~~~~
+
+Pros
+^^^^
+
+* Battle-tested, widely adopted, big community, lots of documentation and answers.
+* Enables event replay-ability.
+
+Cons
+^^^^
+
+* Complex to manage, including likely manual scaling.
+* Simple consumers requires additional code for some messaging features.
+
+Consequences
+------------
+
+* Operators will need to deploy and manage the selected infrastructure, which is likely to be complex. If Apache Kafka is selected, there are likely to be a set of auxiliary parts to provide all required functionality for our message bus. However, third-party hosting is also available (see separate decision).
+* Most of the consequences of an event bus should related to `OEP-52: Event Bus Architecture`_ more generally, and hopefully will not be Kafka specific.
+
+Rejected Alternatives
+---------------------
+
+Apache Pulsar
+~~~~~~~~~~~~~
+
+Although rejected to start, `Apache Pulsar`_ remains an option for those looking for an alternative to Kafka.
+
+Pros
+^^^^
+
+* Ease of scalability (built-in, according to docs).
+* Ease of data retention capabilities.
```

Comment:
'ease of capabilities' is a strange construction. Maybe  `good data retention capabilities` or `easy data retention`